sklearn/linear_model/logistic.py

```diff
@@ -549,10 +560,13 @@ def logistic_regression_path(X, y, pos_class=None, Cs=10, fit_intercept=True,
         # np.unique(y) gives labels in sorted order.
         pos_class = classes[1]
 
+    # If sample weights are not provided, set them to 1 for all examples
+    if sample_weight is None:
+        sample_weight = np.ones(X.shape[0])
+
```

Comment:
you might want to do 

`sample_weight = np.array(sample_weight)` here, to support lists, etc

Also you can add a `check_consistent_length` to check that they are the same size.